hrpc/call.go

```diff
@@ -33,6 +34,7 @@ type RegionInfo interface {
 	Table() []byte
 	SetClient(RegionClient)
 	Client() RegionClient
+	json.Marshaler
```

Comment:
You don't gain much by adding the json.Marshaler interface to the RegionInfo interface. All you really need is for region.info to implement MarshalJSON. So, I suggest you don't add this here. These interfaces are already way too big.